components/plan-build-ps1/bin/hab-plan-build.ps1

```diff
@@ -2155,7 +2155,7 @@ try {
 
     # Test to ensure package name and origin contain only valid characters
     foreach ($var in @("pkg_name", "pkg_origin")) {
-      if (-Not ((Get-Content Variable:\$var) -match '^[A-Za-z0-9_-]+$')) {
+      if (-Not ((Get-Content Variable:\$var) -match '^[A-Za-z0-9_\.-]+$')) {
```

Comment:
Note that this will allow dots in _origin_ names as well, which conflicts with https://github.com/habitat-sh/habitat/blob/1840d65009a009fb17d4561f2aec0ea4bc49f130/components/core/src/package/ident.rs#L14-L17

(Incidentally, this regex was _already_ different than the one in Rust 😱 )

That seems to be outside the scope of what we want to do.

However, I think we need to add whatever validation we do here into the `PackageIdent` struct, as well. (BONUS: When we get around to rewriting the build process in Rust, or at least offloading parts of it to Rust, we'll already have this there :smile:)